codec/src/main/java/io/netty/handler/codec/AsciiString.java

```diff
@@ -34,9 +36,10 @@
  * reduced memory footprint and faster data transfer from/to byte-based data structures such as a byte array and
  * {@link ByteBuf}. It is often used in conjunction with {@link TextHeaders}.
  */
-public final class AsciiString implements CharSequence, Comparable<CharSequence> {
+public final class AsciiString extends ByteString implements CharSequence, Comparable<CharSequence> {
 
     public static final AsciiString EMPTY_STRING = new AsciiString("");
+    public static final byte UPPER_CASE_TO_LOWER_CASE_ASCII_OFFSET = (int) 'a' - (int) 'A';
```

Comment:
DefaultHttp2Headers uses it in its "binary / assumed to be ascii" conversion to lower case.